### PR TITLE
Enums

### PR DESCRIPTION
**Pull request by nezaniel**
This adds enum support and tests for node and reference relation properties

### CODE REVIEW

File: Neos.ContentGraph.PostgreSQLAdapter/src/Domain/Repository/Query/HypergraphChildQuery.php [hunk 1]
```diff
@@ -124,7 +124,7 @@ public function withOnlyTethered(): self
             AND cn.classification = :classification';
 
         $parameters = $this->parameters;
-        $parameters['classification'] = NodeAggregateClassification::CLASSIFICATION_TETHERED;
+        $parameters['classification'] = NodeAggregateClassification::CLASSIFICATION_TETHERED->value;
```
Comment (bwaidelich) [inline Neos.ContentGraph.PostgreSQLAdapter/src/Domain/Repository/Query/HypergraphChildQuery.php:127]: Is that change related to this feature?

**Review by bwaidelich — APPROVED** [review]
Makes sense to me, +1 by reading!

File: Neos.ContentGraph.PostgreSQLAdapter/src/Domain/Repository/Query/HypergraphChildQuery.php (hunk 1 above)
Comment (nezaniel) [reply Neos.ContentGraph.PostgreSQLAdapter/src/Domain/Repository/Query/HypergraphChildQuery.php:127]: nope, but it helps running the tests as it fixes something I seem to have missed

**Review by ahaeslich — APPROVED** [review]
